README.md
cfg2html-all
============


doc/Changelog
2012-04-04 jerome ROBERT <www.admin-sys.com>
  * cfg2html_solaris:
	  Fixed bug align

	* cfg2html_solaris:
	Add metaset

	* cfg2html_solaris:
	Add plugins clustersun.sh	

	* cfg2html_solaris:
	add: svcs -x

	* cfg2html_solaris:
        Fixed bug list_disks

2008-10-03  John Taisto <jr@e-integration>

	* cfg2html_solaris:
	Fixed bug for excludeing zones -- missing in actual 'Z' in getopts.

	* cfg2html_solaris:
	Fixed bug in get_zones function -- Properly handle global zones; can't perform
        zonecfg export on global zones.

	* cfg2html_solaris:
	Fixed bug in get_zones function -- Properly handle if run from a non-global zone.

	* cfg2html_solaris:
        Fixed bug where CFG_PRINTER's default is set to no--although I can see the appeal
        to doing that. ;)

	* cfg2html_solaris:
        Fixed bug where no zpool binary exists and error is generated.  This will
        happen on all solaris version under Solaris 10 update 2 and will happen on
        hosts that we installed without zfs that are Solaris 10 update 2 or newer.

	* Changelog:
	Fixed bug misspelling of changed.

	* cfg2html_solaris:
        + Separated information from different zfs list's by type: filesystem,
        volume, and snapshot.

	* cfg2html_solaris:
        + Modified "vxdisk list" to "vxdisk -o alldgs list"  Reason:
        The disk group column shows imported disk groups as normal and shows
        all other disk groups in parentheses.

	* cfg2html_solaris:
        + Revised the version functionality to display version and the release date of that
        version.

	* cfg2html_solaris:
        + Added the -b option to change the background CCS section of the body of the html
        generated.  You can use this option for customizing your HTML pages background. 
        You can simply change the color as the example in the -h shows or you can change the
        color and add a positioned image with repeat or no-repeat like so:
        -b "#666666 url('company_logo.png') no-repeat fixed top;"

2005-05-19  Ralph Roth  <ralproth@hpul80>

	* cfg2html_sun-VCS.sh:
	fixed Log cvs keyword that contributors changed :)

	* pack.sh, plugins/get_sap.sh, cfg2html_sun.sh:
	+ pack: rcs2log, fixed Log cvs keyword that contributors changed :)

	* pack.sh, plugins/VCS_plugin.sh, plugins/cluster_node_status.ksh:
	New files provided by Keiran Sweet, (version 1.11)

	* cfg2html_sun.sh: SUN collector update by Keiran Sweet

2004-06-03  Ralph Roth  <ralproth@hpul80>

	* cfg2html_sun.sh: Small changes made by MVL

2004-06-02  Ralph Roth  <ralproth@hpul80>

	* plugins/VxVM_collect.sh, plugins/crontab_collect.sh, cfg2html_sun.sh:
	Checked in new version from MVL, send back for testing

2004-04-29  Ralph Roth  <ralproth@hpul80>

	* cfg2html_sun.sh: Send to MVL

2004-04-26  Ralph Roth  <ralproth@hpul80>

	* plugins/VxVM_collect.sh, plugins/crontab_collect.sh, plugins/get_sap.sh:
	+ ripped and added the plugisn from the HPUX stream

	* cfg2html_sun.sh: ! Merged the enhancements from MVL

2004-04-19  Ralph Roth  <ralproth@hpul80>

	* plugins/get_sap.sh: New file.

	* plugins/get_sap.sh: Version 1.6 from MVL
	+ password
	+ printers
	- ibmdpo

	* plugins/get_sap.sh: Onsite Fixes (aw)

	* cfg2html_sun.sh: New file.

	* cfg2html_sun.sh: Version 1.6 from MVL
	+ password
	+ printers
	- ibmdpo



sun/ARRAY_info.sh
#!/bin/ksh
# Small script to examine disk arrays and build a nice HTML page
# showing the array name & the status of the disks a happy disk is green
# a sad disk is red an empty slot is shown as cyan 
# written by Gary Staples Feb 2004 inspired by cfg2html
# please remember to test it befor eyou place onto a LIVE system
# tested on solaris 2.6 with A5000 and D1000
# email grjstaples@clara.net
#
GREEN="#00ff00#"
SYS=$(uname -n)
TIME=$(date '+%D  %H:%M')
rm /tmp/Array_info*
rm /tmp/Array_STATUS.html
ARRAY_LIST=$(/usr/sbin/luxadm probe | grep "WWN:" | cut -d':' -f2|awk '{print  $1}')
for ARRAY_NAME in ${ARRAY_LIST}
do 
FRONT="<tr bgcolor=yellow><td><B>FRONT</B></td>"
REAR="<tr bgcolor=yellow><td><B>REAR</B></td>"
echo "<table border=1 valign=top bordercolor=black width=100%>" >> /tmp/Array_STATUS.html
echo "<tr bgcolor=yellow><td width=30%><B>ARRAY status...${SYS}...${TIME}</B></td><td><B>Array name ${ARRAY_NAME}</B></td></tr>" >> /tmp/Array_STATUS.html
echo "</table>" >> /tmp/Array_STATUS.html
echo "<table border=1 valign=top bordercolor=black width=100%>" >> /tmp/Array_STATUS.html
ENCLOSURE_TYPE=$(/usr/sbin/luxadm display ${ARRAY_NAME} | grep -n SUBSYSTEM | cut -d ':' -f 1)
  if [[ ${ENCLOSURE_TYPE} -eq 12 ]] ; then
     /usr/sbin/luxadm display ${ARRAY_NAME} | head -11 | tail -7 > /tmp/Array_info.out
  elif [[ ${ENCLOSURE_TYPE} -eq 16 ]] ; then
     /usr/sbin/luxadm display ${ARRAY_NAME} | head -15 | tail -11 > /tmp/Array_info.out
  fi
cat /tmp/Array_info.out | sed 's/Not Installed/Empty Array Slot/g' > /tmp/Array_info.wrk
cat /tmp/Array_info.wrk | while read W1 W2 W3 W4 W5 W6 W7 
do
BGCOLOR=red
if [[ ${W3} == "(O.K.)" ]]
then
BGCOLOR=${GREEN}
fi
if [[ ${W2} == "Empty" ]]
then
BGCOLOR=cyan
fi
FRONT=${FRONT}"<td bgcolor=${BGCOLOR}><B>${W2}</B></td>" >> /tmp/Array_STATUS.html
BGCOLOR=red
if [[ ${W6} == "(O.K.)" ]]
then
BGCOLOR=${GREEN}
fi
if [[ ${W5} == "Empty" ]]
then
BGCOLOR=cyan
fi
REAR=${REAR}"<td bgcolor=${BGCOLOR}><B>${W5}</B></td>" >> /tmp/Array_STATUS.html
done
echo ${REAR} >> /tmp/Array_STATUS.html
echo ${FRONT} >> /tmp/Array_STATUS.html
echo "</table>" >> /tmp/Array_STATUS.html
done


sun/ClusterSun.sh
#!/usr/bin/ksh
############################################################################
# Cluster SUN / Oracle Collector for cfg2html
############################################################################
# $Header: /home/CVS/cfg2html_sun/plugins/ClusterSun.sh,v 1.0 2012/04/04 10:02:17 jerome ROBERT Exp $
############################################################################
# $Log: ClusterSun.sh,v $
# Revision 1.0  2012/04/04 10:02:17  jerome ROBERT 
# Initial CVS import
# Initial Cluster collector
############################################################################
# (C)opyright 04.04.2012 by jerome ROBERT (http://www.admin-sys.com)
############################################################################



print 'Cluster\c' ; pkginfo -l SUNWscr |grep VERSION
echo " "
echo "                          --------------------------------------"
echo " "
echo "Gestion du Quorum:"
scstat -q
echo " "
echo "                          --------------------------------------"
echo " "
echo "Information Cluster" 
/usr/cluster/bin/scstat -vvv
echo " "
echo "                          --------------------------------------"
echo " "
echo "Configuration cluster:"
echo " "
if [[ -x /usr/cluster/bin/cluster ]]
then
   /usr/cluster/bin/cluster show -vvv
else
	 [[ -x /usr/cluster/bin/scrgadm ]] && scrgadm -pvvv
fi


html/make_index.sh
#!/usr/bin/ksh

# make_index.sh - make an HTML index for the cfg2html output files
#
# Run this shell script in the directory where your cfg2html files 
# are stored. After running this script, load the allhosts.htm
# file in your browser. This is part of the cfg2html package.
#
# Original version 2.8 by Ralph Roth, 2005/05/09
# Hacked beyond recognition by S^2, Aug-Sep 2008
# (most changes were to make this work for cfg2html_solaris)


OUT=index.htm

echo "make_index for cfg2html" 
echo 
echo "Creating an HTML Index of your cfg2html configuration files..."

cat > $OUT <<_EOF_
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>

   <style type=text/css>

   PRE             {font-family: Lucidia-Console, Courier-New, Courier; font-size: 10pt}
   BODY            {font-family: Arial, Verdana, Helvetica, Sans-serif; font-size: 12pt}
   A               {font-familY: Arial, Verdana, Helvetica, Sans-serif}
   A:link          {text-decoration: none}
   A:visited       {text-decoration: none}
   A:hover         {text-decoration: underline}
   A:active        {color: red; text-decoration: none}

   H1              {font-family: Arial, Verdana, Helvetica, Sans-serif; font-size: 20pt}
   H2              {font-family: Arial, Verdana, Helvetica, Sans-serif; font-size: 14pt}
   H3              {font-family: Arial, Verdana, Helvetica, Sans-serif; font-size: 12pt}
   DIV, P, OL, UL, SPAN, TD
                   {font-family: Arial, Verdana, Helvetica, Sans-serif; font-size: 11pt}
   </style>

</HEAD>
<BODY>
<B>Host List</B>
<HR>
<B>
<pre>
_EOF_


for filename in `(find . -name "*_cfg.html" | sed 's|\./||' | sort -d )`; do

   host=`echo $filename | perl -pe 's|_cfg||; s|\..*$||'`

   echo "Creating Host $host"
   echo "<A HREF=\"$filename\" TARGET=\"info\">$host</A>"  >> $OUT
done

#echo "<p><HR><p>Created: `date +%x-%X`" >>$OUT
echo "<HR>Created: `date +%x-%X`" >>$OUT

cat >> $OUT <<_EOF_
</pre>
</B>
<br>
</BODY>
</HTML>
_EOF_

echo "Hosts collected! Now load $OUT in your browser!"

# end


sun/get_sys_info.sh
#!/bin/ksh

# Silly little script to collect system (OS and hardware) data
#
# For Solaris and Red Hat only.  
#
# S^2 30 Jul 2008

OS=`uname -s`;

if [ $OS = "Linux" ]; then

     #                     /--------------------\
     #                    < Get CPU info (Linux) >
     #                     \--------------------/

     qty=`grep "^physical id" /proc/cpuinfo | sort -u | grep -c physical`
     num_cores=`grep -c "core id" /proc/cpuinfo`

     tmp=/tmp/foo.$$
     dmidecode |\
     sed -n '/^Handle 0x04/,/^Handle 0x07/p' > $tmp
     type=`awk -F': ' '/Manufacturer:/  {mfg = $2} /Family:/ {fam = $2} END {printf "%s-%s\n", mfg, fam}' $tmp`
     spd=`cat $tmp | egrep "Current Speed:" | head -1 | awk -F": " '{print $2}' | awk '{print $1}'`
     tot_threads=$num_cores
     rm -f $tmp

     #                    /-------------------\
     #                   < Get HW Make n Model >
     #                    \-------------------/

     dmidecode |\
     sed -n '/Handle 0x0100/,/^Handle/p' > $tmp
     mfg=`  awk -F': ' '/Manufacturer/ {print $2}' $tmp`;
     model=`awk -F': ' '/Product Name/ {print $2}' $tmp`;
     rm -f $tmp
     #printf "%s\t%s\n" mfg model

     #                    /-------------------\
     #                   < Get OS Version Info >
     #                    \-------------------/

     os_release=`cat /etc/redhat-release | tr -d '(' | tr -d ')' | sed -e 's|Red Hat Enterprise Linux .. |RHEL|' -e 's|release ||' -e 's|Update |u|' | awk '{print $1$3}'`


     #                    /--------------------\
     #                   < Get RAM info (Linux) >
     #                    \--------------------/

     ram_avail=`awk '/MemTotal:/ {printf "%3.0f\n", $2/1015.879}' /proc/meminfo`
     # trim leading spaces
     ram_avail=`echo $ram_avail | awk '{print $1}'`

elif [ $OS = "SunOS" ]; then

     #                       /--------\
     #                      < Solaris! >
     #                       \--------/

     #                     /------------\
     #                    < Get CPU info >
     #                     \------------/

     plat=`uname -p`;
     if   [ $plat = "i386"  ]; then

          #type=`psrinfo -pv | egrep -v 'MHz| virtual ' | awk '{print $1,$(NF-1), $(NF)}' | tr -d \) | uniq -c`;
          qty=` psrinfo -pv | egrep '^The physical processor' | grep -c '^' `

          type=`psrinfo -pv | egrep -v 'x86|virtual' | tail -1 | sed -e 's|     ||' `;
          type=`echo $type | sed -e 's|(tm) Processor||'`
          type=`echo $type | sed -e 's|Dual Core ||'`

          spd=` psrinfo -pv | egrep MHz | tail -1 | awk '{print $(NF-1)}' | tr -d \)`;

          tot_threads=`uname -X | awk '/NumCPU/ {print $3}'`;
          num_cores=$tot_threads;

     elif [ $plat = "sparc" ]; then

          cpu_info=`psrinfo -pv | egrep 'SPARC' | awk '{print $1,$(NF-1), $(NF)}' | tr -d \) | uniq -c`;

          qty=` printf "$cpu_info\n" | awk '{print $1}'`;

          type=`printf "$cpu_info\n" | awk '{print $2}'`;
          spd=` printf "$cpu_info\n" | awk '{print $3}'`;

          # Determine total number of threads for all CPUs on this host
          tot_threads=`uname -X | awk '/NumCPU/ {print $3}'`;
          threads_per_core=1;
          [ $type = 'UltraSPARC-T1'   ] && threads_per_core=4
          [ $type = 'UltraSPARC-T2'   ] && threads_per_core=8

          # Calculate number of cores
          num_cores=$(( tot_threads / threads_per_core ));
     else
          printf "Unrecognized platform, uname -p is neither i386 nor sparc\n";
          exit 1;
     fi

     #                    /----------------------\
     #                   < Get RAM info (Solaris) >
     #                    \----------------------/

     ram_avail=`prtconf | head -2 | nawk -F": " '/Megab/ {print $2}' | nawk '{printf "%3.0f\n", $1}'`
     # trim leading spaces
     ram_avail=`echo $ram_avail | awk '{print $1}'`

     #                    /-------------------\
     #                   < Get OS Version Info >
     #                    \-------------------/

     host_os=`uname -s`
     host_os=`echo $host_os | sed 's/SunOS/Solaris/' `

     host_os_rev=`uname -r`
     host_os_rev=`echo $host_os_rev | sed 's|5\.||' `

     os_ver="$host_os $host_os_rev";
     os_release=`cat /etc/release | grep Solaris | head -1 | sed 's|^ *||g' `;


     #                    /-------------------\
     #                   < Get HW Make n Model >
     #                    \-------------------/

     class=`uname -m`;
     model=`prtdiag | head -1 | cut -c40-99 | perl -pe "s/^\s+//g"| cut -c7-99`;
     [ $plat = "i386" ] && model=`prtdiag | head -1 | cut -c40-99 | perl -pe "s/^\s+//g"`;
     mfg='Sun';
     [ $class = "sun4us" ] && mfg='Fujitsu';
     #printf "%s\t%s\n" "$mfg" "$model";
     [ "$model" = 'Sun Fire T200' ] && model='Sun Fire T2000';
     [ "$model" = 'Sun Fire(TM) T1000' ] && model='Sun Fire T1000';
     #model=`echo $model | sed 's/Sun Fire /SF /' `


else
     printf "Unhandled OS >$OS<, aborting...\n";
     printf "(only know how to deal with Solaris and Linux)\n";

fi

printf "Manufacturer:     %s\n"    "$mfg"; 
printf "Model:            %s\n"    "$model";
printf "Operating System: %s\n"    "$os_release";
printf "CPU(s):           %dx %-15s @%4d MHz; %2d Cores %2d Threads\n" $qty "$type" $spd $num_cores $tot_threads;
printf "Memory:           %s MB\n" "$ram_avail";

# fini


sun/VxVM_collect.sh
############################################################################
# Veritas Volume Manager (VxVM) Collector for cfg2html
############################################################################
# $Header: /home/CVS/cfg2html_sun/plugins/VxVM_collect.sh,v 1.2 2004/06/02 13:52:17 ralproth Exp $
############################################################################
# $Log: VxVM_collect.sh,v $
# Revision 1.2  2004/06/02 13:52:17  ralproth
# Checked in new version from MVL, send back for testing
#
# Revision 1.2  2004/06/02 13:48:42  ralproth
# I tested it to work fine on Solaris 7 and 8 and got good output with VxVM 3.2 and 4.0 (3.5 is alike 4.0 so I expect no hickups there). The bulk of the work went into rearranging the VxVM stuff. In may cases we got double the information, guess that's fixed now.
#
# Revision 1.1  2004/04/26 15:31:40  ralproth
# + ripped and added the plugisn from the HPUX stream
#
# Revision 1.1  2004/04/24 20:22:47  ralproth
# + added plugins from cfg2html form the HPUX package
#
# Revision 2.1.1.1  2003/01/21 10:33:32  ralproth
# Import from HPUX to cygwin
#
# Revision 1.2  2002/02/06 09:10:17  ralproth
# VxVM collector added
#
# Revision 1.1  2002/02/05 12:41:33  ralproth
# Initial CVS import
# Initial VxVM collector
############################################################################
# (C)opyright 04.02.2002 by ROSE SWE, Ralph Roth (rose_swe@hotmail.com)
############################################################################


#for i in `vxdg list |awk '{print ($1)}'|grep -v DEVICE`
#  do
#	echo "Volumegroup $i\n"
#	vxdg list $i
#	done

echo "VxPrint\n"
vxprint -rth

echo "\n"
echo "VxStat"

	vxstat -d 2>&1 | tail +3 | awk '
	    BEGIN { 
		printf ("                                OPERATIONS             BLOCKS       AVG TIME(ms)\n");
		printf ("TYP NAME                      READ     WRITE       READ      WRITE  READ  WRITE\n");
	     }
    		{
		    v  = $1
		    n  = $2
		    or = $3
		    ow = $4
		    br = $5
		    bw = $6
		    ar = $7
		    aw = $8
		    printf ("%s %-20s %9s %9s %10s %10s %5.1f  %5.1f\n", v,n,or,ow,br,bw,ar,aw)

		}'                             

